internal/provider/models/team.go

```diff
@@ -64,3 +80,55 @@ func (data *Team) ReadFromResponse(ctx context.Context, team *iam.Team) diag.Dia
 
 	return nil
 }
+
+func (data *TeamResource) ReadFromResponse(ctx context.Context, team *iam.Team, memberIds *[]string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	data.Id = types.StringValue(team.Id)
+	data.Name = types.StringValue(team.Name)
+	if team.Description != nil && *team.Description != "" {
+		data.Description = types.StringValue(*team.Description)
+	} else {
+		data.Description = types.StringNull()
+	}
+	if memberIds != nil && len(*memberIds) > 0 {
+		data.MemberIds, diags = utils.StringSet(memberIds)
+		if diags.HasError() {
+			return diags
+		}
+	} else {
+		data.MemberIds = types.SetNull(types.StringType)
+	}
+	data.IsIdpManaged = types.BoolValue(team.IsIdpManaged)
+	orgRole := string(team.OrganizationRole)
+	if orgRole != "" && orgRole != string(iam.ORGANIZATIONMEMBER) {
+		data.OrganizationRole = types.StringValue(orgRole)
+	} else {
+		data.OrganizationRole = types.StringNull()
+	}
```

Comment:
wrote it this way because i kept hitting `Provider produced inconsistent result after apply` and because the API sets org role to org member automatically if org role is not provided